module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,56 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    for (const auto& line : bitmap) {
+        uint8_t count{};
+        uint8_t pixel = line.at(0);
+        for (const auto& el : line) {
+            if (pixel == el) {
+                ++count;
+            } else {
+                compressed.push_back(std::make_pair(pixel, count));
+                count = 1;
+                pixel = el;
+            }
+        }
+        compressed.push_back(std::make_pair(pixel, count));
+    }
+
+    return compressed;
+}
+
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(const std::vector<std::pair<uint8_t, uint8_t>>& compressed) {
+    std::array<std::array<uint8_t, width>, height> bitmap;
+
+    size_t count{}, line{}, el{};
```

Comment:
Every declaration/ definition in separate line.

Reply:
Decompress function re-implemented with one iterator to make is simpler and reduce the number of variables. Two iterators used in STL#2 and inner loop reduced in favour of stl algorithm.